### DISCUSSION TITLE
New Gantt parsing error

### DISCUSSION THREAD

**Discussion by mstoc** [Giammaria/Vega-Visuals#6] [Q&A]
The new Gantt looks great! However I am getting the error "Expression parse error: +utcFormat(timeOffset('day', timeOffset('month', toDate(utcFormat(datum['startDate'], '%m')+'/01/'+utcFormat(datum['startDate'], '%Y')), 1),-1), '%d')". Do you know what might be causing this?

**Comment by Giammaria** [comment]
Hey @mstoc, A separate user had a similar issue. Can you confirm that your root node has an `id`, and that your second level nodes have that id set as their `parentId`?

**Reply by mstoc** [reply]
Yep, my data looks practically identical to your example

**Reply by Giammaria** [reply]
Awesome. Can you confirm that the fields are set to the following data types?
id - Whole number
parentId - Whole number
name - Text
**startDate - Date
endDate - Date**
decimalPercentComplete - Decimal number
dependencyId - Text
color - Text

Also, can you confirm that none of the Summarization for all fields are set to "Don't summarize"?

**Reply by mstoc** [reply]
Yes those all match

**Reply by Giammaria** [reply]
Are you able to provide a .pbix with data that you are allowed to share that has the same issue, so I can troubleshoot?

**Reply by mstoc** [reply]
I can't drop a pbix in this reply, is there another way I could share it? I also suspect part of the issue of the gantt not showing up is also because of the field specification mapping is not making connections for some columns.

**Reply by Giammaria** [reply]
You don't have this option available, or will it not allow you to attach a .pbix?
![image](https://github.com/user-attachments/assets/acf4ec2b-24a5-464f-81a1-448f4d35aefc)

**Reply by mstoc** [reply]
Cannot attach pbix unfortunately

**Reply by Giammaria** [reply]
Can you email it to giammariam@gmail.com?

**Reply by Giammaria** [reply]
Hi @mstoc. I didn't have a ton of time, but noticed that some of the fields weren't mapping correctly because the first letter of each of their names was capitalized. 

Additionally, I need to fix the dependencyId field so that it knows how to deal with empty strings. In the meantime, set that field to either null in power query or BLANK() in DAX if you don't have a need for dependency lines. 

There is some other weirdness going on with expand/collapse. I wonder if that has something to do with the parent/child ids. I need to dive in further, but have had limited bandwidth today.
